src/engraving/libmscore/measure.cpp

```diff
@@ -4823,4 +4824,238 @@ void Measure::computeMinWidth()
 
     computeMinWidth(s, x, isSystemHeader);
 }
+
+void Measure::layoutSegmentsInPracticeMode(const std::vector<int>& visibleParts)
+{
+    layoutSegmentsWithDuration(visibleParts);
+}
+
+qreal Measure::computeFirstSegmentXPosition(Segment* segment)
+{
+    qreal x{ 0 };
+
+    Shape ls(RectF(0.0, 0.0, 0.0, spatium() * 4));
+
+    x = segment->minLeft(ls);
+
+    if (segment->isChordRestType()) {
+        x += score()->styleP(hasAccidental(segment) ? Sid::barAccidentalDistance : Sid::barNoteDistance);
+    } else if (segment->isClefType() || segment->isHeaderClefType()) {
+        x += score()->styleP(Sid::clefLeftMargin);
+    } else if (segment->isKeySigType()) {
+        x = qMax(x, score()->styleP(Sid::keysigLeftMargin));
+    } else if (segment->isTimeSigType()) {
+        x = qMax(x, score()->styleP(Sid::timesigLeftMargin));
+    }
+    x += segment->extraLeadingSpace().val() * spatium();
+
+    return x;
+}
+
+static Segment* findFirstEnabledSegment(Measure* measure)
+{
+    Segment* current = nullptr;
+    for (current = measure->first(); current && !current->enabled(); current = current->next()) {
+    }
```

Comment:
Empty `for` loop?

Reply:
it chanegs the current segment

Reply:
Yes, I see that it used the side effect, but that's confusing code, to me at least.

Reply:
I agree with JoJo, I believe needs rewrite this code

Reply:
yes, that while loop is better